### ISSUE TITLE
After upgraded to 1.11.8, new bottleneck appears

### ISSUE THREAD

**Issue by Caoxuyang** [paritytech/parity-ethereum#9255]
> I'm running:
>
> - **Which Parity version?**: 1.11.8 (1.11.5 before that)
> - **Which operating system?**:  Linux
> - **How installed?**:  from source
> - **Are you fully synchronized?**:  no
> - **Which network are you connected to?**: private chain
> - **Did you try to restart the node?**: yes

I developed a benchmark tools to test TPS performance on Parity private chain.
Before I upgrade to 1.11.8, I was using 1.11.5. And my benchmark tests went our fine.
Tps can go above 3000.

My test goes like this:

 I usually start 20 processes and send totallly 200000 signed txs to the network and calculate TPS from the blocks. The network consists of one transaction node to receive RPC requests and two mining nodes. 
All the nodes are running in separate VMs. 
My gasLimit is big enough for more than 50000 transactions to be mined in a single block.

But in 1.11.8, I run the exactly same test mentioned above.
Tps is limited by the mining-node. (tx-node has the same behavior, cpu, memory, throughput ...)

In each block, at most 4096 txs are mined. (One transaction uses 21000 gas.)
And I will paste my configure parameters for parity:

`--gasprice 0 --tx-queue-size 400000 --tx-queue-per-sender=25000 --jsonrpc-threads=8 --jsonrpc-server-threads=8`
And my gasLimit is big enough, well, my previous tests proved that.

I looked through all the releases from 1.11.5 to 1.11.8, but I didn't get any idea.
Any help will be great, thanks in advance.

**Comment by Tbaut** [comment]
This looks like an interesting project, thanks for building it.
What is your actual question?

**Comment by Caoxuyang** [comment]
@Tbaut Question is everything works fine with parity v 1.11.5. (TPS was around 3000.)
But after I upgrade my parity to 1.11.8, the performance of mining node seems to appear a new bottleneck.
In each block, there are at most 4096 txs mined. (TPS is only around 800 now)
I wonder if you guys added some new tags to restrict the performance?

**Comment by debris** [comment]
@tomusdrw were there any changes to tx queue that could affect this?

**Comment by tomusdrw** [comment]
Yes: https://github.com/paritytech/parity-ethereum/blob/master/ethcore/sync/src/chain/mod.rs#L155

We only propagate up to 4096 transactions from the queue, so the mining nodes just don't have more transactions. I suppose we should make this number dynamic (dependent on the block gas limit) and propagate up to `BLOCKS * block_gas_limit / 21_000` transactions from the pool, with `BLOCKS =  4 | 5`.

Also it would be good to implement additional propagation strategy, where each new transaction (or at least every local transaction) in the pool is propagated right away (1tx per packet) - currently propagation is triggered every second afair.

**Comment by tomusdrw** [comment]
@Caoxuyang yes, although have a look at the submitted PR (#9265) it should solve the issue and bring the perf back. Let me know the tests results :)

**Comment by Caoxuyang** [comment]
@tomusdrw I tested on 1.11.8. It's cool and the result stays the same, around 3000 tps. Thanks a million.

However I meet another issue, if I set the tx-queue-size to a large number like 500000. 
And I send 500000 transactions to the transaction node, after like 200000 txs mined, the transaction node will automatically disconnect with the mining nodes. The transaction node is still running but the transaction node just can't join the network anymore. (Mining nodes are still connected with each other.) This issue happened on both 1.11.5 and 1.11.8.

**Comment by tomusdrw** [comment]
Could you run the transaction node with `-lsync=trace` and send us some logs from the time when it happens?

**Comment by Caoxuyang** [comment]
@tomusdrw I opened the logs actually, and just now I tried again to reproduce it. Here is the logs:

`2018-08-08 09:05:28 UTC Verifier #0 INFO import  Imported #18 0xf5d6…dc0d (1 txs, 0.07 Mgas, 1 ms, 0.73 KiB)

2018-08-08 09:05:29 UTC IO Worker #1 TRACE sync  Skipping warp sync. State: Idle
2018-08-08 09:05:29 UTC IO Worker #0 DEBUG network  Connecting peers: 2 sessions, 4 pending + 1 started

2018-08-08 09:05:29 UTC IO Worker #1 DEBUG sync  Error sending packet: Error(OversizedPacket, State { next_error: None, backtrace: None })

2018-08-08 09:05:29 UTC IO Worker #1 TRACE sync  02 <- Transactions (108994 entries)
2018-08-08 09:05:29 UTC IO Worker #1 TRACE sync  07 <- Transactions (64 entries)
2018-08-08 09:05:29 UTC IO Worker #1 DEBUG sync  Sent up to 108994 transactions to 2 peers.
2018-08-08 09:05:29 UTC IO Worker #0 TRACE sync  == Disconnecting 2: Parity/v1.11.5-beta-0487c5b-20180629/x86_64-linux-gnu/rustc1.27.0

2018-08-08 09:05:29 UTC IO Worker #0 DEBUG sync  Disconnected 2
2018-08-08 09:05:29 UTC IO Worker #0 TRACE sync  Syncing with peers: 1 active, 1 confirmed, 1 total
2018-08-08 09:05:29 UTC IO Worker #0 TRACE sync  Considering peer 7, force=false, td=None, our td=6125082604576892342340742933771060947139, latest=0xf5d6…dc0d, have_latest=true, state=Idle`

I think the problem is the Error inside, OversizedPacket. However all of the transaction I sent are of the same content and size. 
And thank you for your quick reply!
